server/index.js

```diff
@@ -116,16 +143,32 @@ app.delete('/delete_package', (req, res) => {
       code 200: succesfully returns the list of packages
 */ 
 app.get('/get_packages', (req, res) => {
-  axios({
-    method: 'get', 
-    url: 'https://api.pkge.net/v1/packages/list', 
-    headers: {
-      'Accept': 'application/json',
-      'X-Api-Key': process.env.PKGE_API_KEY
+  const userID = req.query.userId;
+  if(userID == undefined){
+    res.status(400).send("Requires a user ID!");
+  }
+  auth.getAuth().getUser(userID).then((user) => {
+    pkgeAPI.getPackages().then(async (r) => {
+      let packages = {};
+      const snapshot = await db.collection(COLLECTION_NAME).doc(userID).collection('trackings').get();
+      for(const el in r){
+        snapshot.forEach((doc) => {
+          if(doc.id == r[el]['track_number']){
+            packages[r[el]['track_number']] = r[el];
+          }
+        })
+      }
```

Comment:
At first glance, it was hard to know what r and el are right away. Maybe use clearer variable names

Reply:
updated